backend/app/processing/llm.py

```
@@ -0,0 +1,369 @@
+        "product", "buy", "purchase", "price", "cost", "how much", 
+        "where can i get", "looking for", "search for", "find", 
+        "show me", "available", "stock", "shipping", "delivery",
+        "shop", "discount", "deal", "offer", "sale", "coupon"
+    ]
+    return any(indicator in query.lower() for indicator in product_indicators)
+
+def estimate_token_count(text: str) -> int:
+    """Estimate the number of tokens in a text"""
+    return len(text) // 4
+
+def ask_groq(prompt: str, model="llama3-70b-8192") -> str:
```

Ruff (0.8.2)

Missing type annotation for function argument `model`

(ANN001)